src/App.tsx

```diff
@@ -17,18 +78,32 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                onQueryChange={handleQueryChange}
+                onStatusChange={handleStatusChange}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loading && <Loader />}
+              <TodoList
+                todos={getFilteredTodos(todos)}
+                currentTodo={currentTodo}
+                onTodoChose={setCurrentTodo}
+              />
```

Comment:
It might be better to hide the `TodoList` component when the loading isn't completed yet.

Reply:
+